### PR TITLE
chore(react-radio): Update Radio indicator for perf and alignment at 125% DPI

### LINKED ISSUE #30025: [Bug]: Radio glitch: Active radio indicator is not perfectly centered
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
Windows 11 / Edge
```

### Are you reporting Accessibility issue?

no

### Reproduction

https://react.fluentui.dev/?path=/docs/components-radiogroup--default

### Bug Description

Change system display scale to 125%

## Actual Behavior
Radio selection indicator is slightly misaligned

## Expected Behavior
Radio selection indactor is perfectly centered

image

### Logs

_No response_

### Requested priority

Normal

### Products/sites affected

_No response_

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### LINKED ISSUE #30210: [Bug]: Disabled Radio controls don't appear disabled in high contrast (forced-colors)
### Environment Information

- **Package version(s)**: 9
- **Browser and OS versions**: Edge, Windows

#### Actual behavior:

Radio buttons don't use the disabled colors in system high contrast mode (forced-colors).

screenshot of disabled radio buttons in high contrast

#### Expected behavior:

Disabled Radio buttons should use the GrayText color in high contrast.

### If applicable, please provide a codepen repro:

https://react.fluentui.dev/?path=/docs/components-radiogroup--default#disabled

### Priorities and help requested (not applicable if asking question):

Are you willing to submit a PR to fix? Yes

Requested priority: Normal

Products/sites affected: (if applicable)

### PR DESCRIPTION
**Pull request by behowell**
## Previous Behavior

Radio used a circle SVG icon to show selection by default. This had a number of issues:
* Using a SVG for a simple circle adds unnecessary bundle size and rendering complexity.
* Hiding the SVG for unchecked radios used a selector that ended in `> *`, which is known to be bad for CSS performance.
* Browser pixel rounding caused the indicator to not be centered at 125% DPI.
* Disabled Radio buttons weren't using the `GrayText` color in high contrast (forced-colors).

## New Behavior

* Replace the `CircleFilled` icon with an `::after` pseudo-element to show the checked state, by default.
   * This pseudo-element is only rendered on **checked** radios (unlike the SVG which was rendered on all radios at opacity 0).
   * Remove the much-maligned `> *` selector for the checked state when the default indicator is used.
* For backwards compatibility, Radio still supports a custom icon passed into the `indicator` slot, and falls back to the original styles in that case.
* Use `scale()` to set the size of the checked indicator, to work around browser pixel rounding errors.
* Use `GrayText` colors for forced colors mode when disabled.

## Related Issue(s)

- Fixes #30025
- Fixes #30210

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30207/perf-test-react-components/Avatar-mount_base.html">642</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30207/perf-test-react-components/Avatar-mount.html">634</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30207/perf-test-react-components/Button-mount_base.html">308</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30207/perf-test-react-components/Button-mount.html">311</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30207/perf-test-react-components/Field-mount_base.html">1183</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30207/perf-test-react-components/Field-mount.html">1139</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30207/perf-test-react-components/FluentProvider-mount_base.html">706</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30207/perf-test-react-components/FluentProvider-mount.html">737</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30207/perf-test-react-components/FluentProviderWithTheme-mount_base.html">78</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30207/perf-test-react-components/FluentProviderWithTheme-mount.html">86</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30207/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">66</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30207/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">76</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30207/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">76</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30207/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">86</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30207/perf-test-react-components/MakeStyles-mount_base.html">892</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30207/perf-test-react-components/MakeStyles-mount.html">844</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30207/perf-test-react-components/Persona-mount_base.html">1752</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30207/perf-test-react-components/Persona-mount.html">1768</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30207/perf-test-react-components/SpinButton-mount_base.html">1384</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30207/perf-test-react-components/SpinButton-mount.html">1405</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                                     |      Baseline (minified/GZIP) |                            PR |                                                                                                                                                                                                     Change |
| :-------------------------------------------------------------------------------------------------------------------- | ----------------------------: | ----------------------------: | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-radio</samp> <br /> <abbr title='bundle-size/Radio.fixture.js'>Radio</abbr>                               |  `31.738 kB`<br />`10.365 kB` |    `30.84 kB`<br />`9.953 kB` | `-898 B` <br />`-412 B`  |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/DataGrid.fixture.js'>DataGrid</abbr>                         | `154.888 kB`<br />`43.533 kB` | `155.142 kB`<br />`43.629 kB` |    `254 B` <br />`96 B`  |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableAsDataGrid.fixture.js'>Table as DataGrid</abbr>         | `128.109 kB`<br />`34.792 kB` | `128.362 kB`<br />`34.883 kB` |    `253 B` <br />`91 B`  |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableSelectionOnly.fixture.js'>Table (Selection only)</abbr> |  `73.288 kB`<br />`20.017 kB` |  `73.542 kB`<br />`20.106 kB` |    `254 B` <br />`89 B`  |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableSortOnly.fixture.js'>Table (Sort only)</abbr>           |  `71.895 kB`<br />`19.617 kB` |  `72.149 kB`<br />`19.708 kB` |    `254 B` <br />`91 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              |  `69.893 kB`<br />`20.261 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `209.769 kB`<br />`59.924 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |   `42.38 kB`<br />`14.104 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |    `7.099 kB`<br />`2.385 kB` |
| <samp>react-radio</samp> <br /> <abbr title='bundle-size/RadioGroup.fixture.js'>RadioGroup</abbr>                                                                             |   `15.337 kB`<br />`6.263 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/PrimitivesOnly.fixture.js'>Table (Primitives only)</abbr>                                                            |  `43.891 kB`<br />`13.782 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/2a42b52a62aa5bc6e2c0fb81e43d4edf0b0fc4e9'>2a42b52a62aa5bc6e2c0fb81e43d4edf0b0fc4e9</a></sub>
<div id="bundleSizeComment9423"/>

File: apps/vr-tests-react-components/src/stories/Radio.stories.tsx [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import * as React from 'react';
```
Comment (fabricteam) [inline apps/vr-tests-react-components/src/stories/Radio.stories.tsx:1]: ## [🕵 fluentuiv9 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=30207&commitId=775057c67ee763f43b79fbcc824178c86a1d2103&env=prod&clientType=FLUENTUI)


  <details><summary><b>Radio Converged</b> 23 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Radio Converged.checked - Dark Mode.active.chromium.png|2| Changed|
|Radio Converged.checked - Dark Mode.hover.chromium.png|3| Changed|
|Radio Converged.checked.hover.chromium.png|4| Changed|
|Radio Converged.disabled+checked.default.chromium.png|2| Changed|
|Radio Converged.group-horizontal-stacked - RTL.default.chromium.png|4| Changed|
|Radio Converged.checked - Dark Mode.rest.chromium.png|2| Changed|
|Radio Converged.checked.rest.chromium.png|4| Changed|
|Radio Converged.group-horizontal.default.chromium.png|4| Changed|
|Radio Converged.group-vertical - RTL.default.chromium.png|4| Changed|
|Radio Converged.checked - High Contrast.hover.chromium.png|4| Changed|
|Radio Converged.checked.active.chromium.png|5| Changed|
|Radio Converged.checked - High Contrast.active.chromium.png|4| Changed|
|Radio Converged.group-horizontal - RTL.default.chromium.png|4| Changed|
|Radio Converged.checked - High Contrast.rest.chromium.png|4| Changed|
|Radio Converged.disabled+checked - High Contrast.default.chromium.png|4| Changed|
|Radio Converged.group-horizontal-stacked.default.chromium.png|4| Changed|
|Radio Converged.group-vertical.default.chromium.png|4| Changed|
|Radio Converged.group-disabled - RTL.default.chromium.png|2| Changed|
|Radio Converged.group-disabled.default.chromium.png|2| Changed|
|Radio Converged.indicator+checked.active.chromium.png|0| Added|
|Radio Converged.indicator+checked.hover.chromium.png|0| Added|
|Radio Converged.indicator+checked.rest.chromium.png|0| Added|
|Radio Converged.indicator+unchecked.default.chromium.png|0| Added|

</details>


  <details><summary><b>Table flex - subtle selection</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Table flex - subtle selection.rest.hover unselected row.chromium.png|4| Changed|
|Table flex - subtle selection.rest.chromium.png|4| Changed|

</details>


  <details><summary><b>Table layout flex</b> 11 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Table layout flex.single select (checked) - brand - Dark Mode.chromium.png|3| Changed|
|Table layout flex.single select (checked) - brand - High Contrast.chromium.png|4| Changed|
|Table layout flex.single select (checked).chromium.png|4| Changed|
|Table layout flex.single select (checked) - brand - RTL.chromium.png|4| Changed|
|Table layout flex.single select (checked) - neutral.chromium.png|3| Changed|
|Table layout flex.single select (checked) - Dark Mode.chromium.png|2| Changed|
|Table layout flex.single select (checked) - High Contrast.chromium.png|4| Changed|
|Table layout flex.single select (checked) - neutral - Dark Mode.chromium.png|2| Changed|
|Table layout flex.single select (checked) - brand.chromium.png|4| Changed|
|Table layout flex.single select (checked) - neutral - RTL.chromium.png|3| Changed|
|Table layout flex.single select (checked) - RTL.chromium.png|4| Changed|

</details>


  <details><summary><b>Table layout table</b> 11 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Table layout table.single select (checked) - RTL.chromium.png|4| Changed|
|Table layout table.single select (checked) - High Contrast.chromium.png|4| Changed|
|Table layout table.single select (checked) - brand - RTL.chromium.png|4| Changed|
|Table layout table.single select (checked) - brand.chromium.png|4| Changed|
|Table layout table.single select (checked) - brand - High Contrast.chromium.png|4| Changed|
|Table layout table.single select (checked) - neutral - RTL.chromium.png|3| Changed|
|Table layout table.single select (checked) - Dark Mode.chromium.png|2| Changed|
|Table layout table.single select (checked).chromium.png|4| Changed|
|Table layout table.single select (checked) - neutral.chromium.png|3| Changed|
|Table layout table.single select (checked) - brand - Dark Mode.chromium.png|3| Changed|
|Table layout table.single select (checked) - neutral - Dark Mode.chromium.png|2| Changed|

</details>


  <details><summary><b>Table table - subtle selection</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Table table - subtle selection.rest.chromium.png|4| Changed|
|Table table - subtle selection.rest.hover unselected row.chromium.png|4| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

Comment (spmonahan) [reply apps/vr-tests-react-components/src/stories/Radio.stories.tsx:1]: These look good to me but a second set of eyes is probably a good thing here.

**Review by spmonahan — APPROVED** [review]
> Remove the much-maligned > * selector for the checked state.

Unfortunately, it's still added even though we don't use in the default case anymore :(

Still, nice update! As a 125% display scaler I appreciate this :)

**Comment by behowell** [comment]
> > Remove the much-maligned > * selector for the checked state.
> 
> Unfortunately, it's still added even though we don't use in the default case anymore :(
> 
> Still, nice update! As a 125% display scaler I appreciate this :)

To clarify, it's only added if the Radio has a custom indicator icon; it's not in the styles by default like it used to be. I don't know of any cases where someone uses a custom icon, but it would be a breaking change to stop supporting that, so I had to leave support for that case.

File: apps/vr-tests-react-components/src/stories/Radio.stories.tsx (hunk 1 above)
Comment (behowell) [reply apps/vr-tests-react-components/src/stories/Radio.stories.tsx:1]: Changes LGTM as well. The new indicator circle is a fraction of a pixel larger, but its radius is still ~10px, which is what is in the figma.

**Review by ling1726 — APPROVED** [review]

**Pull request merged**